### DISCUSSION TITLE
Cannot pass script args to xpm run

### DISCUSSION THREAD

**Discussion by AaronFontaine-DojoFive** [xpack/xpm-js#140] [Q&A]
Note: If this is not the correct place for this question, I can move it.  I checked out Tapatalk but it says the forum is deprecated in favor of GitHub Discussions and the threads appear locked.  There is nothing else posted here, so hopefully this is the correct place.

The synopsis for xpm run is:  
`xpm run [options...] [<command>] [-- <script args>]`

I would like to be able to pass extra optional arguments to the CMake and/or Ninja commands directly from the command line.  I have tried everything I can to get `<script args>` to pass through, but no matter what I do I get "error: optional arguments not supported for array of commands"

I have tried having the action defined in both the `scripts` and the `actions` properties.  I have made sure that the elements are not JSON arrays (i.e. no opening or closing square brackets).  I have tried `xpm run-script` in place of `xpm run`.  I'm not understanding how we pass extra parameters to the underlying commands without having all possible permutations spelled out in the `package.json`.

For example, in the Hello World Quick Start template, we have a single `test` command under `scripts`.  I tried testing with this:  
```
$ xpm run test -- "testing"
error: optional arguments not supported for array of commands

$ xpm run-script test -- "testing"
error: optional arguments not supported for array of commands
```

Also, the `clean` command of both the Debug and Release configurations is a single command and not an array of commands.  This also does not work:  
```
$ xpm run clean --config Debug -- "testing"
error: optional arguments not supported for array of commands
```

What am I missing?

**Comment by ilg-ul** [comment]
Hi Aaron,

Thank you for your perseverance in testing this. You are not missing anything, actions with a single command should accept extra arguments. I checked and it does not, so it must be a bug (#141).

**Reply by ilg-ul** [reply]
@aaronf-at-d5, please test the 0.13.6 pre-release (xpm@next) and let me know if it works, so I can proceed with the release.
